File: sales2/domain/usecase/customer/DeleteCustomerUseCase.kt
package edu.itvo.sales2.domain.usecase.customer

import edu.itvo.sales2.domain.repository.CustomerRepository
import javax.inject.Inject

class DeleteCustomerUseCase @Inject constructor(
    private val repository: CustomerRepository
) {
    suspend operator fun invoke(customerId: String){
        val existing = repository.findCustomerById(customerId)

        require(existing!=null){
            "Customer with id: $customerId does not exist"
        }
    }
}

File: webserviceT/application/use-cases/customer/delete-customer.usecase.ts
import type { CustomerRepository } from "../../../src/domain/repositories/customer.repository";

export class DeleteCustomerUseCase {
    constructor(private repository: CustomerRepository) {}

    async execute(id: string) {
        if (!id) {
            throw new Error("Customer code is required");
        }

        const existing = await this.repository.findById(id);

        if (!existing) {
            throw new Error("Customer not found");
        }

        return this.repository.delete(id);
    }
}